app/src/forms/email/reminderService.js

```diff
@@ -0,0 +1,296 @@
+        late: 0
+      });
+    }
+
+    // check for the current period
+    for (let i = 0; i <dates.length; i++) {
+      let startDate = moment(dates[i].startDate).format('YYYY-MM-DD');
+      let graceDate = (late) ? moment(dates[i].graceDate).format('YYYY-MM-DD') : moment(dates[i].closeDate).format('YYYY-MM-DD') ;
+      if(toDay.isBetween(startDate, graceDate)) {
+        return Object({
+          state : 1,
+          index: i ,
+          dates: dates[i],
+          old_dates : (i==0) ? null :  dates[i-1],
+          late: (toDay.isBetween(dates[i].closeDate, dates[i].graceDate))? 1 : 0
+        });
+
+      }
+    }
+
+    let first = dates[0];
+    return Object({
+      state: (toDay.isBefore(first.startDate)) ? -1 : 0,
+      index: -1,
+      dates: false,
+      old_dates : false,
+      late: -1
+    });
+
+  },
+  _listDates: (schedule) =>{
+
+    if (schedule.scheduleType == ScheduleType.PERIOD){
+      return getAvailableDates(
+        schedule.keepOpenForTerm,
+        schedule.keepOpenForInterval,
+        schedule.openSubmissionDateTime,
+        schedule.repeatSubmission.everyTerm,
+        schedule.repeatSubmission.everyIntervalType,
+        schedule.allowLateSubmissions.forNext.term,
+        schedule.allowLateSubmissions.forNext.intervalType,
+        schedule.repeatSubmission.repeatUntil,
+        schedule.scheduleType,
+        schedule.closeSubmissionDateTime
+      );
+    }
+    if (schedule.scheduleType == ScheduleType.MANUAL){
+      return [Object({
+        startDate: schedule.openSubmissionDateTime,
+        closeDate:  null,
+        graceDate:  null
+      })];
+    }
+
+    if (schedule.scheduleType == ScheduleType.CLOSINGDATE){
+      return [ Object({
+        startDate: schedule.openSubmissionDateTime,
+        closeDate: schedule.closeSubmissionDateTime,
+        graceDate: service._getGraceDate(schedule)
+      })];
+    }
+
+  },
+  _getGraceDate: (schedule)=>{
+    let substartDate = moment(schedule.openSubmissionDateTime);
+    let newDate = substartDate.clone();
+    return (schedule.allowLateSubmissions.enabled) ?   newDate.add(schedule.allowLateSubmissions.forNext.term , schedule.allowLateSubmissions.forNext.intervalType).format('YYYY-MM-DD HH:MM:SS') : null ;
+  },
+  _getForms: async ()=> {
+    let fs = [];
+    await Form.query()
+      .modify('reminderEnabled')
+      .modify('filterActive', true)
+      .then(forms => {
+        fs = forms;
+      });
+    return fs;
+  },
+  _getReminders : async (forms)=>{
+    let reminder = [];
+    let toDay = moment();
+    for (let i = 0; i< forms.length; i++) {
+      let obj = {};
+
+      obj.avalaibleDate =  service._listDates(forms[i].schedule) ;
+
+      if (obj.avalaibleDate.length == 0) {
+        reminder.push({ error:true, message : `Form ${forms[i].name } has no avalaible date.` });
+        continue;
+      }
+
+      obj.report = service.getCurrentPeriod(obj.avalaibleDate, toDay, forms[i].schedule.allowLateSubmissions.enabled);
```

Comment:
availableDate is spelled incorrectly, it should be `obj.availableDate`

Reply:
It was done last week

Reply:
available is spelled incorrectly, it should be available

Reply:
It seems to have been added back in then